include/astshim/detail/utils.h

```diff
@@ -112,6 +113,28 @@ inline std::string stringToUpper(std::string const &str) {
     return upstr;
 }
 
+/**
+An mutable variable-length char manages memory using RAII
+
+Note: this class unnecessary in C++17 because std::string data is mutable; use
+    std::string foo;
+    foo.resize(nchar);
+    foo.data() is your safe, mutable char * array
+instead of:
+    SafeCStr foo(nchar);
+    foo.array is your safe mutable char * array
+*/
+class SafeCString {
+public:
+    char *array;
+    SafeCString(std::size_t nchar) : array(nullptr) { array = new char[nchar]; }
+    ~SafeCString() {
+        if (array != nullptr) {
+            delete[] array;
+        }
+    }
+};
```

Comment:
This is not safe at present; if you copy this object, `array` will get deleted twice.

I think the best way to fix that is to use `unique_ptr<char[]>` (see http://en.cppreference.com/w/cpp/memory/unique_ptr/operator_at), which is probably simple enough that you don't need a separate class.

Reply:
Excellent suggestion. I implemented that. Please review again.